_code-samples/issue-credentials/js/issuer_service.js

```diff
@@ -0,0 +1,207 @@
+  app.post("/credential", async (req, res) => {
+    try {
+      // validateCredentialRequest() throws if the request is not validly formatted
+      const credRequest = validateCredentialRequest(req.body);
+
+      /**
+       * As a credential issuer, you typically need to verify some information
+       * about someone before you issue them a credential. For this example,
+       * the user passes relevant information in a documents field of the API request.
+       * The documents are kept confidential, off-chain.
+       * 
+       * verifyDocuments() throws if the provided documents don't pass inspection
+       */
+      verifyDocuments(req.body);
+
+      const credXrpl = credentialToXrpl(credRequest);
+
+      const tx = {
+        TransactionType: "CredentialCreate",
+        Account: wallet.address,
+        Subject: credXrpl.subject,
+        CredentialType: credXrpl.credential,
+        URI: credXrpl.uri,
+        Expiration: credXrpl.expiration,
+      };
+      const ccResponse = await client.submit(tx, { autofill: true, wallet });
+
+      if (ccResponse.result.engine_result === "tecDUPLICATE") {
+        throw new XRPLTxError(ccResponse, 409);
+      } else if (ccResponse.result.engine_result !== "tesSUCCESS") {
+        throw new XRPLTxError(ccResponse);
+      }
+
+      return res.status(201).json(ccResponse.result);
+    } catch (err) {
+      return handleAppError(res, err);
+    }
+  });
+
+  // GET /admin/credential - Method for admins to look up all credentials issued -------------------
+  app.get("/admin/credential", async (req, res) => {
+    try {
+      // ?accepted=yes|no|both query parameter - the default is "both"
+      const query = Object.fromEntries(
+        Object.entries(req.query).map(([k, v]) => [k.toLowerCase(), v])
+      );
+      const filterAccepted = (query.accepted || "both").toLowerCase();
+
+      const credentials = await lookUpCredentials(client, wallet.address, "", filterAccepted);
+      const result = credentials.map((entry) => credentialFromXrpl(entry));
+
+      return res.status(200).json({ credentials: result });
+    } catch (err) {
+      return handleAppError(res, err);
+    }
+  });
+
+  // DELETE /admin/credential - Method for admins to revoke an issued credential ----------------------------
+  app.delete("/admin/credential", async (req, res) => {
+    let delRequest;
+    try {
+      delRequest = validateCredentialRequest(req.body);
+      const { credential } = credentialToXrpl(delRequest);
+
+      // To save on transaction fees, check if the credential exists on ledger before attempting to delete it.
+      // If the credential is not found, a RippledError (`entryNotFound`) is thrown.
+      await client.request({
+        command: "ledger_entry",
+        credential: {
+          subject: delRequest.subject,
+          issuer: wallet.address,
+          credential_type: credential,
+        },
```

Comment:
It is recommended to specify the ledger_index explicitly as per the [doc](https://xrpl.org/docs/references/protocol/data-types/basic-data-types#specifying-ledgers)
```suggestion
        },
        ledger_index: "validated",
```